state/factory/workingset.go

```diff
@@ -53,7 +53,7 @@ type (
 		writeviewFunc func(name string, v interface{}) error
 		dbFunc        func() db.KVStore
 		delStateFunc  func(string, []byte) error
-		statesFunc    func(opts ...protocol.StateOption) (uint64, state.Iterator, error)
```

Comment:
basically, this PR replaces the current `FilterOption` with `KeysOption`, think no need to change here